src/environment.cpp

```diff
@@ -623,7 +622,7 @@ void ServerEnvironment::savePlayer(RemotePlayer *player)
 	player->save(players_path, m_gamedef);
 }
 
-RemotePlayer *ServerEnvironment::loadPlayer(const std::string &playername)
+RemotePlayer *ServerEnvironment::loadPlayer(const std::string &playername, PlayerSAO *sao)
```

Comment:
Why pass this to the method, and not just use the `getPlayerSAO` on player, which you generate below anyway?